src/custom-elements/src/components/date/date.tsx

```diff
@@ -287,6 +297,272 @@ export class Date {
     this._updateViewMonth();
   }
 
+  componentDidRender(): void {
+    if (this._afterRenderHandler instanceof Function) {
+      this._afterRenderHandler();
+      this._afterRenderHandler = null;
+      return
+    }
+
+    if (!this._keyboardFocusedDate) {
+      const today = dayjs().format(this.format);
+      this._keyboardFocusedDate = this._getValidDayElement(this.value)
+        ? this.value
+        : this._getValidDayElement(today)
+          ? today
+          : this._getFirstOrLastAvailableDate(true);
+    }
+  }
+
+  componentDidLoad(): void {
+    document.body.addEventListener('keydown', this._onKeyDown.bind(this));
+    document.body.addEventListener('focusin', this._onFocusIn.bind(this));
+  }
+
+  componentDidUnload(): void {
+    document.body.removeEventListener('keydown', this._onKeyDown);
+    document.body.removeEventListener('focusin', this._onFocusIn);
+  }
+
+  _onFocusIn(e) {
+    if (e.target !== document.body && e.target !== null) {
+      this._enableKeyboardNavigation = contains(this.el, e.target);
+    }
+  }
+
+  /**
+  * Handles key down event
+  */
+  _onKeyDown(e: KeyboardEvent) {
+    // TODO: check if calendar is focused in to enable arrows
+    if (!this._enableKeyboardNavigation || !contains(this.el, e.target as HTMLElement)) {
+      return;
+    }
+
+    const keyHandler = {
+      "ArrowLeft": this._focusSiblingDay.bind(this, -1),
+      "ArrowRight": this._focusSiblingDay.bind(this, 1),
+      "ArrowUp": this._focusSiblingWeek.bind(this, -1),
+      "ArrowDown": this._focusSiblingWeek.bind(this, 1),
+      "Tab": this._handleTab.bind(this, e),
+      "Enter": this._handleEnter.bind(this, e),
+      "Escape": this._handleEscape.bind(this)
+    }[e.key];
+
+    const disableArrowHandler = e.key.startsWith("Arrow") && !this._isTargetCalendarDay(e);
+    if (!keyHandler || disableArrowHandler || !contains(this.el, e.target as HTMLElement)) {
+      return;
+    }
+
+    e.stopPropagation();
+    e.preventDefault();
+    keyHandler();
+  }
+
+  /**
+   * Checks wether the keyboard event target belongs to a calendar day
+  */
+  _isTargetCalendarDay(e: KeyboardEvent) {
+    return (e.target as HTMLElement).classList.contains("chi-datepicker__day");
```

Comment:
Do we have this class in the `classes.ts` file? If not include it and same for the rest.

```suggestion
    return (e.target as HTMLElement).classList.contains(DATEPICKER_DAY);
```